### PR TITLE
vuln: historic bl memory exposure vulnerability

### PR DESCRIPTION
**Pull request by ChALkeR**
Similar to 309, but I fixed the vector.

/cc @feross — is the data correct?

### CODE REVIEW

**Comment by feross** [comment]
LGTM

**Comment by mcollina** [comment]
Side note, I tried. bl 0.8 depends on an old version of readable-stream, which uses the Buffer ctor but its dev environment is not so easy to restore (tests are not passing in modern node.js). Fixing this is hard.

File: vuln/npm/391.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "author": "Feross Aboukhadijeh",
+  "module_name": "bl",
+  "created_at": "2017-04-24",
+  "updated_at": "2017-04-24",
+  "publish_date": "2016-01-19",
+  "cves": [],
+  "vulnerable_versions": "<=0.9.4 || 1.0.0",
+  "patched_versions": ">=1.0.1 || >=0.9.5 <1.0.0",
+  "slug": "bl-memory-exposure",
+  "overview": "bl.append(number) in the affected `bl` versions passes a number to Buffer constructor, appeding a chunk of unitialized memory",
```
Comment (Trott) [inline vuln/npm/391.json]: Not sure if we want to fix small typos in this file or if they don't really matter, but if we want to fix: `appeding` -> `appending`

Comment (ChALkeR) [reply vuln/npm/391.json]: I think that fixing typos makes sense, as we often modify those files in fact with various fixes. So they are not frozen.

**Pull request merged**